### PR TITLE
doc: Update how-to-run-kata-containers-with-SE-VMs.md

### PR DESCRIPTION
**Pull request by BbolroC**
This PR updates the documentation `how-to-run-kata-containers-with-SE-VMs.md` with the following changes:

- Update secure boot image verification for IBM SE
- Remove unnecessary `sudo`
- Add an error message where an incorrect host key document is used
- Add a missing artifact `kernel-confidential-modules`
- Make a variable `kernel_version` and replace it with relevant hits

Signed-off-by: Hyounggyu Choi <Hyounggyu.Choi@ibm.com>

### CODE REVIEW

File: docs/how-to/how-to-run-kata-containers-with-SE-VMs.md [hunk 1]
```diff
@@ -88,19 +88,19 @@ However, if any of these components are absent, they must be built from the
...
 $ tar -tf build/kata-static-rootfs-initrd-confidential.tar.xz | grep initrd
 ./opt/kata/share/kata-containers/kata-containers-initrd-confidential.img
 ./opt/kata/share/kata-containers/kata-ubuntu-20.04-confidential.initrd
 $ mkdir artifacts
-$ tar -xvf build/kata-static-kernel-confidential.tar.xz -C artifacts ./opt/kata/share/kata-containers/vmlinuz-6.1.62-121-confidential
+$ tar -xvf build/kata-static-kernel-confidential.tar.xz -C artifacts ./opt/kata/share/kata-containers/vmlinuz-${kernel_version}-confidential
 $ tar -xvf build/kata-static-rootfs-initrd-confidential.tar.xz -C artifacts ./opt/kata/share/kata-containers/kata-ubuntu-20.04-confidential.initrd
 $ ls artifacts/opt/kata/share/kata-containers/
-kata-ubuntu-20.04-confidential.initrd  vmlinuz-6.1.62-121-confidential
+kata-ubuntu-20.04-confidential.initrd  vmlinuz-6.7-136-confidential
```
Comment (GabyCT) [inline docs/how-to/how-to-run-kata-containers-with-SE-VMs.md:103]: probably replace it by the kernel version variable

**Review by GabyCT — APPROVED** [review]

File: docs/how-to/how-to-run-kata-containers-with-SE-VMs.md (hunk 1 above)
Comment (BbolroC) [reply docs/how-to/how-to-run-kata-containers-with-SE-VMs.md:103]: oh, good catch! I will update it together with a reference fix when https://github.com/confidential-containers/confidential-containers/pull/241 is merged. Thanks!

**Review by stevenhorsman — APPROVED** [review]
Apart from Gaby's suggestions, the changes look okay to me. Thanks!

**Comment by BbolroC** [comment]
As mentioned above, a link at https://github.com/kata-containers/kata-containers/blob/dfcb41b5cc782fdffb1356d80dd203881ea8b3e8/docs/how-to/how-to-run-kata-containers-with-SE-VMs.md?plain=1#L321 should be updated when the documentation is migrated from operator to confidential-containers. Thanks.

File: docs/how-to/how-to-run-kata-containers-with-SE-VMs.md (hunk 1 above)
Comment (BbolroC) [reply docs/how-to/how-to-run-kata-containers-with-SE-VMs.md]: done! 😉

**Pull request merged**